queue/src/main/java/org/killbill/notificationq/NotificationQueueDispatcher.java

```diff
@@ -164,17 +166,18 @@ public boolean isStarted() {
     }
 
     @Override
-    public int doProcessEvents() {
+    public DispatchResultMetrics doDispatchEvents() {
         final List<NotificationEventModelDao> notifications = getReadyNotifications();
         if (notifications.isEmpty()) {
-            return 0;
+            return new DispatchResultMetrics(0, -1);
         }
         log.debug("Notifications from {} to process: {}", config.getTableName(), notifications);
 
         for (final NotificationEventModelDao cur : notifications) {
             dispatcher.dispatch(cur);
         }
-        return notifications.size();
+        // No need to return time, this is easy to compute from caller
+        return new DispatchResultMetrics(notifications.size(), -1);
```

Comment:
Returning `-1` is a bit confusing though because below we call `dispatchTime.update(metricsResult.getTimeNanoSec(), TimeUnit.NANOSECONDS);` regardless of the value. It works because notification isn't sticky but it's a bit sneaky...

Reply:
Yes i agree, but i would then need to do another change of interface to pass the metrics data from`NotificationQueueDispatcher#getReadyNotifications()`, and since i could easily avoid propagating this data because it's easy to compute in the client side, i made this 'hack'. I am also not so happy about it - but thought it was not worth creating more java classes, ... to pass this data.